test/e2e/tests/test_hpo.py

```diff
@@ -61,26 +76,102 @@ def get_sagemaker_hpo_job(sagemaker_client, hpo_job_name):
         )
         return None
 
+def get_hpo_sagemaker_status(hpo_job_name: str):
+    hpo_sm_desc = get_sagemaker_hpo_job(hpo_job_name)
+    return hpo_sm_desc["HyperParameterTuningJobStatus"]
+
+def get_hpo_resource_status(reference: k8s.CustomResourceReference):
+    resource = k8s.get_resource(reference)
+    assert "hyperParameterTuningJobStatus" in resource["status"]
+    return resource["status"]["hyperParameterTuningJobStatus"]
 
 @service_marker
 @pytest.mark.canary
 class TestHPO:
-    def test_create_hpo(self, sagemaker_client, xgboost_hpojob):
+    list_status_created = ("InProgress", "Completed")
+    list_status_stopped = ("Stopped", "Stopping")
+    status_inprogress: str = "InProgress"
+    status_completed: str = "Completed"
+
+    def _wait_resource_hpo_status(
+        self,
+        reference: k8s.CustomResourceReference,
+        expected_status: str,
+        wait_periods: int = 30,
+        period_length: int = 30,
+    ):
+        return wait_for_status(
+        expected_status, wait_periods, period_length, get_hpo_resource_status, reference
+    )
+
+    def _wait_sagemaker_hpo_status(
+        self, 
+        hpo_job_name,
+        expected_status: str,
+        wait_periods: int = 30,
+        period_length: int = 30,
+    ):
+        return wait_for_status(
+            expected_status,
+            wait_periods,
+            period_length,
+            get_hpo_sagemaker_status,
+            hpo_job_name,
+        )
+
+    def _assert_hpo_status_in_sync(
+        self, hpo_job_name, reference, expected_status
+    ):
+        assert (
+            self._wait_sagemaker_hpo_status(
+                hpo_job_name, expected_status
+            )
+            == self._wait_resource_hpo_status(reference, expected_status)
+            == expected_status
+        )
+
+    def test_hpo(self, xgboost_hpojob):
+        (reference, resource) = xgboost_hpojob
+        assert k8s.get_resource_exists(reference)
+    
+        hpo_job_name = resource["spec"].get("hyperParameterTuningJobName", None)
+        assert hpo_job_name is not None
+
+        hpo_sm_desc = get_sagemaker_hpo_job(hpo_job_name)
+        assert k8s.get_resource_arn(resource) == hpo_sm_desc["HyperParameterTuningJobArn"]
+        assert hpo_sm_desc["HyperParameterTuningJobStatus"] in self.list_status_created
```

Comment:
for all tests, I think we should remove `Completed` from `list_status_created`  because immidiately after this we are checking for `"ACK.ResourceSynced", "False"` which is contradictory

`Completed` should actually be part of `list_status_stopped` because job can actually go from `Stopping` -> `Completed` if it was stopped when it was almost finished(like model was being uploaded)

Reply:
@mbaijal @surajkota why are we deleting the Kubernetes CR when a job "completes"? I thought we only wanted to delete jobs when a user explicitly deleted the job CR in Kubernetes?

Reply:
I think there is some miscommunication, we are testing the case where the user deletes the CR after the job is complete, controller does not delete the CR automatically

Sounds good ?